flink-table/flink-sql-parser-hive/src/main/java/org/apache/flink/sql/parser/hive/ddl/HiveDDLUtils.java

```diff
@@ -317,4 +321,54 @@ public static SqlTableColumn deepCopyTableColumn(SqlTableColumn column) {
 				column.getParserPosition()
 		);
 	}
+
+	// the input of sql-client will escape '\', unescape it so that users can write hive dialect
+	public static void unescapeProperties(SqlNodeList properties) {
+		if (properties != null) {
+			for (int i = 0; i < properties.size(); i++) {
+				SqlNode node = properties.get(i);
+				// for properties
+				if (node instanceof SqlTableOption) {
+					node = unescapeTableOption((SqlTableOption) node);
+					properties.set(i, node);
+				}
+			}
+		}
+	}
+
+	public static SqlTableOption unescapeTableOption(SqlTableOption option) {
+		String key = StringEscapeUtils.unescapeJava(option.getKeyString());
+		String val = StringEscapeUtils.unescapeJava(option.getValueString());
+		SqlNode keyNode = SqlLiteral.createCharString(key, option.getKey().getParserPosition());
```

Comment:
I think we only need to unescape the option value.

Reply:
Both key and value can have special characters in Hive.